File: simple-pos-management-app/client/src/types/category.ts
import { Product } from './product';

export interface CategoryBase {
  name: string;
  status: boolean;
}

export interface Category extends CategoryBase {
  id: number;
  created_at: string;  
  updated_at: string;
}

export type CategoryCreate = CategoryBase;

export interface CategoryWithProducts extends Category {
  products: Product[];
}

File: simple-pos-management-app/server/app/schemas/category.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, TYPE_CHECKING

if TYPE_CHECKING:
    from app.schemas.product import Product


class CategoryBase(BaseModel):
    name: str
    status: bool = True


class CategoryCreate(CategoryBase):
    pass


class Category(CategoryBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class CategoryWithProducts(Category):
    products: List["Product"] = []

    class Config:
        from_attributes = True
